wandb/integration/keras_core/model_checkpoint.py

```diff
@@ -0,0 +1,173 @@
+import os
+from typing import Dict, List, Literal, Optional, Union
+
+import wandb
+from wandb.sdk.lib import telemetry
+from wandb.sdk.lib.paths import StrPath
+
+try:
+    from keras_core.callbacks import ModelCheckpoint
+except Exception as e:
+    wandb.Error(e)
+
+
+Mode = Literal["auto", "min", "max"]
+SaveStrategy = Literal["epoch"]
+
+
+def _log_artifact(
+    filepath: StrPath,
+    artifact_type: str,
+    aliases: Optional[List] = None,
+    metadata: Optional[Dict] = None,
+) -> None:
+    """Log an artifact to Weights & Biases."""
+    aliases = ["latest"] if aliases is None else aliases + ["latest"]
+    run_configs = wandb.run.config.as_dict()
+    metadata = run_configs if metadata is None else {**metadata, **run_configs}
+    model_checkpoint_artifact = wandb.Artifact(
+        f"run_{wandb.run.id}_model", type=artifact_type, metadata=metadata
+    )
+    if os.path.isfile(filepath):
```

Comment:
I also recommend updating this logic because this might not work. The file path is what we provide to the callback. If `save_weights_only` is True or `save_best_only` is True and a different `monitor` (other than default) is used `FileNotFoundError` is thrown. The base `ModelCheckpoint` callback works fine. 

Check out this PR: https://github.com/wandb/wandb/pull/6423